purepyvmware/base_connector.py

```diff
@@ -0,0 +1,130 @@
+                                                                 verify_ssl)
+        if self.vc_instance:
+            self.sms_instance = BaseConnector.connect_sms_env(self, vcenter_ip)
+
+    def connect_purestorage_fa(self, target, username, password, api_token, verify_https):
+        """Create a session (verified or unverified) with the requested FlashArray.
+
+        Args:
+            target (str): FlashArray IP address or FQDN.
+            username (str): Authenticated user for FlashArray.
+            password (str): Password for FlashArray authenticated user.
+            api_token (str): Authenticated API token for FlashArray.
+            verify_https (bool): Whether or not the session should be verified.
+
+        Returns:
+            fa_instance (purestorage.FlashArray): Verified or unverified session to FlashArray.
+        """
+        # If end-user is using CLI and doesn't want to type their password in clear text on screen they can use this
+        # option to input their password.
+        if not (api_token or password):
+            password = getpass.getpass(prompt='FlashArray Password: ', stream=None)
+
+        if not verify_https:
+            urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+            fa_instance = FlashArray(target, username=username, password=password, api_token=api_token,
+                                     verify_https=verify_https)
+        else:
+            fa_instance = FlashArray(target, username, password, api_token)
+
+        return fa_instance
+
+    def connect_sms_env(self, vcenter_ip):
+        """Create a session (verified or unverified) with the SMS service on the requested vCenter Server.
+
+        Args:
+            vcenter_ip (str): IP address of the vCenter Server you wish to connect to the SMS service on.
+
+        Returns:
+            sms_instance (sms.ServiceInstance): Service Instance for the SMS service on vCenter.
+        """
+        client_stub = self.vc_instance._GetStub()
+        ssl_context = client_stub.schemeArgs.get('context')
+        session_cookie = client_stub.cookie.split('"')[1]
+        additional_headers = {'vcSessionCookie': session_cookie}
+        stub = connect.SoapStubAdapter(vcenter_ip, path='/sms/sdk', version='sms.version.version5',
+                                       sslContext=ssl_context, requestContext=additional_headers)
+        sms_instance = sms.ServiceInstance('ServiceInstance', stub)
+
+        return sms_instance
+
+    def connect_vsphere_env(self, host, user, pwd, sslContext):
+        """Create a session (verified or unverified) with the requested vCenter Server.
+
+        Args:
+            host (str): vCenter Server IP address or FQDN.
+            user (str): Authenticated User for vCenter Server.
+            pwd (str): Password for vCenter authenticated user.
+            sslContext (bool): Whether or not the session should be verified.
```

Comment:
This type of case is how javascript does it, but not in Python.

You would want this to be something like `ssl_context`

Reply:
also, as above provide a default for these. It seems like there's a lot of areas where we allow/ignore SSL/HTTPS/etc. maybe allow setting it once so it applies everywhere?